bin/send_email.py

```diff
@@ -6,60 +6,150 @@
 from typing import List
 
 
-def get_participant_id(syn: synapseclient.Synapse, submission_id: str) -> List[str]:
-    """
-    Retrieves the teamId of the participating team that made
-    the submission. If the submitter is an individual rather than
-    a team, the userId for the individual is retrieved.
-
-    Arguments:
-      syn: A Synapse Python client instance
-      submission_id: The ID for an individual submission within an evaluation queue
-
-    Returns:
-      Returns the synID of a team or individual participant
-    """
-    # Retrieve a Submission object
-    submission = syn.getSubmission(submission_id, downloadFile=False)
-
-    # Get the teamId or userId of submitter
-    participant_id = submission.get("teamId") or submission.get("userId")
-
-    # Ensure that the participant_id returned is a list
-    # so it can be fed into syn.sendMessage(...) later.
-    return [participant_id]
-
-
-def send_email(view_id: str, submission_id: str):
-    """
-    Sends an e-mail on the status of the individual submission
-    to the participant team or participant individual.
-
-    Arguments:
-      view_id: The view Id of the Submission View on Synapse
-      submission_id: The ID for an individual submission within an evaluation queue
-    """
-    syn = synapseclient.login()
-    status = syn.getSubmissionStatus(submission_id)["submissionAnnotations"]["validation_status"]
-
-    # Get the synapse users to send an e-mail to
-    ids_to_notify = get_participant_id(syn, submission_id)
-
-    # Sends an e-mail notifying participant(s) that the evaluation succeeded
-    if status == ["VALIDATED"]:
-      subject = f"Evaluation Success: {submission_id}"
-      body = f"Submission {submission_id} has been evaluated. View your scores here: https://www.synapse.org/#!Synapse:{view_id}/tables/"
-
-    # Otherwise, send an error message to participant(s) and engineers of the infrastructure
-    else:
-      subject = f"Evaluation Failed: {submission_id}"
-      body = f"Evaluation failed for Submission {submission_id}. Submission was left with a validation status of {status}. View your submissions here: https://www.synapse.org/#!Synapse:{view_id}/tables/"
-
-    syn.sendMessage(userIds=ids_to_notify,
-                    messageSubject=subject,
-                    messageBody=body)
+class SendEmail:
```

Comment:
I moved all the functions under a class for 2 reason:

1. For cleanup
2. There is a method in here called `get_participant_id` which gets called in another script, and I plan to later use class inheritance to call that method (once I refactor that script later).

Reply:
Unless there are other reasons you wanted to use inheritence to call `get_participant_id` from `build_update_subfolders.py`, you could instead make it a `staticmethod` so you don't have to instantiate the class to use it or just a standalone function that can get imported wherever it is needed.

Reply:
Consider the differences in Functional Programming vs Object-Oriented Programming:
https://www.datacamp.com/tutorial/functional-programming-vs-object-oriented-programming

I'm on-board with how you'd like this to end up looking like, however, this may lend itself better to a functional approach where each method you've written is stand-alone functionality that could be independently called.

Reply:
Hi all, thanks for the comments.

Thanks @BryanFauble for the link; after doing some reading and looking at the review comments, I'm in agreement with your take on this being better suited for FP since it's for a single functionality (sending the e-mail), and because of the way the functions are written (i.e. as a series of non-destructive operations with defined inputs and outputs). Let me know if this was your rationale here so I can get a better understanding. Either way I made the restructuring in my latest commit. Thanks!

Reply:
Yeah that's the idea here. Generally I'd recommend following functional programming if the intended result of your actions is to kick off a series of processes or bring together semi-related data.

On the other hand OOP is very useful when managing the state of your instances is the pivotal piece.